client/my-sites/site-settings/date-time-format/date-time-format-options.jsx

```diff
@@ -0,0 +1,337 @@
+								? translate( 'Preview: %s', {
+									args: now.format( phpToMomentDatetimeFormat( dateFormat ) ),
+									comment: 'Date/time format preview',
+								} )
+								: ''
+							}
+						</FormSettingExplanation>
+					</span>
+				</FormLabel>
+			</FormFieldset>
+		);
+	}
+
+	timeFormatOption() {
+		const {
+			fields: { time_format: timeFormat },
+			isRequestingSettings,
+			translate,
+		} = this.props;
+		const { customTimeFormat: isCustomFormat } = this.state;
+
+		const now = this.getNow();
+
+		return (
+			<FormFieldset>
+				<FormLabel>
+					{ translate( 'Time Format' ) }
+				</FormLabel>
+				{ defaultTimeFormats.map( ( format, key ) =>
+					<FormLabel key={ key }>
+						<FormRadio
+							checked={ ! isCustomFormat && format === timeFormat }
+							disabled={ isRequestingSettings }
+							name="time_format"
+							onChange={ this.setTimeFormat }
+							value={ format }
+						/>
+						<span>{ now.format( phpToMomentDatetimeFormat( format ) ) }</span>
+					</FormLabel>
+				) }
+				<FormLabel className="date-time-format__custom-field">
+					<FormRadio
+						checked={ isCustomFormat }
+						disabled={ isRequestingSettings }
+						name="time_format"
+						onChange={ this.setCustomTimeFormat }
+						value={ timeFormat }
+					/>
+					<span>
+						{ translate( 'Custom', { comment: 'Custom date/time format field' } ) }
+						<FormInput
+							disabled={ isRequestingSettings }
+							name="time_format_custom"
+							onChange={ this.setCustomTimeFormat }
+							type="text"
+							value={ timeFormat || '' }
+						/>
+						<FormSettingExplanation>
+							{ isCustomFormat && timeFormat
+								? translate( 'Preview: %s', {
+									args: now.format( phpToMomentDatetimeFormat( timeFormat ) ),
+									comment: 'Date/time format preview'
+								} )
+								: ''
+							}
+						</FormSettingExplanation>
+					</span>
+					<FormSettingExplanation>
+						<ExternalLink
+							href="https://codex.wordpress.org/Formatting_Date_and_Time"
+							icon
+							target="_blank"
+						>
+							{ translate( 'Learn more about date and time formatting.' ) }
+						</ExternalLink>
+					</FormSettingExplanation>
+				</FormLabel>
+			</FormFieldset>
+		);
+	}
+
+	startOfWeekOption() {
+		const {
+			fields: { start_of_week: startOfWeek },
+			handleSelect,
+			isRequestingSettings,
+			translate,
+		} = this.props;
+
+		const daysOfWeek = [
+			translate( 'Sunday' ),
+			translate( 'Monday' ),
+			translate( 'Tuesday' ),
+			translate( 'Wednesday' ),
+			translate( 'Thursday' ),
+			translate( 'Friday' ),
+			translate( 'Saturday' ),
+		];
+
+		return (
+			<FormFieldset>
+				<FormLabel>
+					{ translate( 'Week starts on' ) }
+				</FormLabel>
+				<FormSelect
+					disabled={ isRequestingSettings }
+					name="start_of_week"
+					onChange={ handleSelect }
+					value={ startOfWeek || 0 }
+				>
+					{ daysOfWeek.map( ( day, index ) =>
+						<option key={ index } value={ index }>
```

Comment:
a valid use of `index` as `key` but the translated day string might be just as valid